doc/index.md

```diff
@@ -119,27 +197,28 @@ Retrieves the next complete headers object (i.e. a block of headers or trailers)
 
 ### `stream:write_headers(headers, end_stream, timeout)` <!-- --> {#stream:write_headers}
 
-Write the given [`headers`](#http.headers) object to the stream.
-Takes a flag indicating if this is the last chunk in the stream, if `true` the stream will be closed.
+Write the given [`headers`](#http.headers) object to the stream. The function takes a flag indicating if this is the last chunk in the stream, if `true` the stream will be closed. If `timeout` is specified, the socket will be polled untile `timeout` is exceeded.
```

Comment:
"will be polled" => not exactly... it's the maximum amount of time to wait for the data to get sent out from the OS. this may include connection completion time (which may include DNS, TLS, etc), or alternatively if the TCP outgoing it will be waiting for it to drain.